crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -328,7 +334,7 @@ where
 
     fn specialize(
         &self,
-        key: Self::Key,
+        mut key: Self::Key,
```

Comment:
Making this mutable looks suspicious to me because the specialisation key is used with the pipeline caching _before_ specialisation is done? But I’d have to re-read the code around it.

Reply:
I think the code is what I wanted, it's just a bit confusing now that you mention it.

We _do_ want DEPTH_CLAMP_ORTHO to be in the key, which tells us that we want unclipped depth.

But after the key is set, we _may or may not_ want the DEPTH_CLAMP_ORTHO shader define in the prepass shader, depending on whether or not the GPU natively supports unclipped depth.

Modifying the key in this function is just so that the rest of the logic in it skips the DEPTH_CLAMP_ORTHO shader workaround we have.

I think I'll rename DEPTH_CLAMP_ORTHO to UNCLIPPED_DEPTH_ORTHO or something.

Reply:
Right, good point about the key field expressing the user intent, but the possible problem is that other specialisation functions will build on the pipeline descriptor returned from this and they will not see the change to the key. Maybe it doesn’t matter in practice currently but it feels like it could. So what I mean is - maybe another layer of specialisation looks at the key, sees depth clamp ortho is set, and does something based on that assuming the current behaviour (say, presence of the interpolated unclamped position as fragment shader input)?

Reply:
I just saw the changed version. That indeed looks like a good approach to me - differentiate between the intent in the key and the intent in the shader def. Good stuff. This would need a migration doc comment in the description for the change in shader key and shader def. Which is a good thing in this case. :)

Reply:
Yeah I'll write a migration guide once I've tested perf and made sure I didn't break anything.

Glad that the new version is clearer.